crates/re_viewport/src/viewport.rs

```diff
@@ -120,31 +157,29 @@ impl<'a, 'b> Viewport<'a, 'b> {
             return;
         }
 
+        let mut maximized = blueprint.maximized;
+
         if let Some(space_view_id) = blueprint.maximized {
             if !blueprint.space_views.contains_key(&space_view_id) {
-                blueprint.maximized = None; // protect against bad deserialized data
+                maximized = None;
             } else if let Some(tile_id) = blueprint.tree.tiles.find_pane(&space_view_id) {
                 if !blueprint.tree.tiles.is_visible(tile_id) {
-                    blueprint.maximized = None; // Automatically de-maximize views that aren't visible anymore.
+                    maximized = None;
                 }
             }
         }
 
         let mut maximized_tree;
 
+        // TODO(jleibs): This tree won't have the edits from `viewport_blueprint_ui`.
+        // Maybe we should route that all the way through to here and only save it once.
```

Comment:
not entirely following this one. The problem is that the tree doesn't contain potential changes from viewport_blueprint_ui which will be applied next frame? I doubt that's a big issue given that we work with frame delayed state in a lot of places, meaning that this just behaves like everything else. But feels like I'm missing something!

Reply:
Bad stale comment from a previous iteration where this was more problematic.